docs/Routing.md

```diff
@@ -0,0 +1,157 @@
+# Routing
+
+[<< Back to index](./index.md)
+
+1. [FirstSpirit Navigation Service](#firstspirit-navigation-service)
+2. [Basic Routing Principles](#basic-routing-principles)
+3. [The triggerRouteChange Function](#the-triggerroutechange-function)
+
+## FirstSpirit Navigation Service
+
+The FSXA-PWA directly maps the navigation structure of the underlying FirstSpirit project into the predefined navigation component. This is achieved using the navigation service which maps all of the routes as defined in FirstSpirit to CaaS URLs. The data is fetched by the [FSXA-API](https://github.com/e-Spirit/fsxa-api). Here is an example of navigation data.
+
+```json
+  "c8a158a3-2ba3-427c-a7e4-7d41d9844464": {
+      "id": "c8a158a3-2ba3-427c-a7e4-7d41d9844464",
+      "parentIds": [
+          "5a7cdf48-5031-4fcd-b6c7-99e802d0ce57"
+      ],
+      "label": "Home",
+      "contentReference": "https://url.to.your.content",
+      "caasDocumentId": "c8a158a3-2ba3-427c-a7e4-7d41d9844464",
+      "seoRoute": "/Home/",
+      "seoRouteRegex": null,
+      "customData": null
+  },
+  "aa684dc6-2220-4d71-b468-f3e056b0c4f0": {
+      "id": "aa684dc6-2220-4d71-b468-f3e056b0c4f0",
+      "parentIds": [],
+      "label": "Our Solutions",
+      "contentReference": "https://url.to.your.content",
+      "caasDocumentId": "4bd4e5af-c55d-4b46-b335-58e284f2f4c6",
+      "seoRoute": "/Our-Solutions/",
+      "seoRouteRegex": null,
+      "customData": null
+  },
+```
+
+It contains labels and contentReferences which you can use to fill a navigation menu or sitemap.
+
+## Basic Routing Principles
+
+In Nuxt.js [routing](https://nuxtjs.org/docs/2.x/get-started/routing) works via convention. Usually you can link to any pages you may have created in the `pages` folder using the `Nuxt-Link` component. It has a prop called `to` that takes a route as parameter. If you have subfolders in the `pages` folder you can add them to the route as well. You can also use this feature to override pages defined by the navigation service by adding a page with the same pageId as an existing page.
+
+```html
+<NuxtLink to="/">Home page</NuxtLink>
+<NuxtLink to="/myPage">Links to MyPage.vue in the pages folder </NuxtLink>
+<NuxtLink to="/aboutPages/myAboutPage">Links to MyAboutPage.vue in the pages/aboutPages folder.</NuxtLink>
+```
+
+However since routing is slightly more complex in the context of the FSXA-PWA, we recommend using the [triggerRouteChange function](the-triggerroutechange-function) instead of the `<nuxt-link>` component. Using this function, you can still define your own pages and override existing ones. With the added advantage of being able to link to pages in the navigation structure and switching the locale.
```

Comment:
The link to the `triggerRouteChange function` does not works. I guess there is a `h` missing.

Reply:
Actually we recommend using the `href` or `to` attribute **and** the triggerRouteChange function.
We could either use the `a`-tag or the `nuxt-link` component. It would still use the triggerRouteChange function, but we provide an url for the case when JavaScript is disabled.

Reply:
Then the search for "how do I nuxt-link in tsx?" continues